### PR TITLE
Adds minions, dynamic build script and config merge on restart

### PR DESCRIPTION
**Pull request by SecMeyo**
Changes:
- adds build script for easy deployment with CD pipelines
- adds a minion to the stack
- adds ability to change server.config.yaml on stack restart without loosing certs
- several small fixes and improvements

Removes:
- ability to build image with docker compose

### CODE REVIEW

**Comment by weslambert** [comment]
Thanks for the contribution(s)! I haven't had as much time as I would like to optimize this (was primarily a proof-of-concept for a while), so I really appreciate your work!  I'll review this soon. In the meantime, would you mind updating the PR to include how the building/deployment of the container/image might change (ex. `README`)?

**Comment by SecMeyo** [comment]
yep, will do the readme tomorrow

**Comment by SecMeyo** [comment]
Readme is done. I also improved the build script to provide proper info with --help and take all input as parameters with proper flags

File: docker-compose.yaml [hunk 1]
```diff
@@ -14,8 +11,26 @@ services:
...
+      - VELOX_API_HOSTNAME=velociraptor
     ports:
       - "8000:8000"
       - "8001:8001"
       - "8889:8889"
-    restart: unless-stopped
+    networks:
+      - velo
+
+  velociraptor_minion:
```
Comment (weslambert) [inline docker-compose.yaml:22]: We might want to make the usage of a client container optional. We should consider using something like the following:

```
container_name: velociraptor_minion
profiles: 
      - minion
```

Then `docker-compose` can be invoked like:

`docker-compose --profile minion up`

**Comment by SecMeyo** [comment]
the last commit should resolve your comment and improve the readme :)

**Comment by SecMeyo** [comment]
did some happy path testing today and figured that the minion was not connecting successfully to the master. The following was tested and works

:heavy_check_mark: `docker compose up`
:heavy_check_mark: `docker compose --profile minion up`
:heavy_check_mark: `docker compose up` with preexisting server.config.yaml not causing keys to change

**Comment by hakkabara** [comment]
the PR seems to be dead right?

**Comment by SecMeyo** [comment]
Ye, was never merged because Wes and I are simply too short on time

**Comment by weslambert** [comment]
I really appreciate it, but it's been a while since this was originally proposed. I suspect some things may have changed since. I am closing this issue for now due to age. Please feel free to re-open or re-submit with amended changes to accommodate the original goal. Thanks!

**Pull request closed without merging**